### PR TITLE
Restructure readme

### PR DESCRIPTION
**Pull request by JesseMK**
> This is a draft modify since I make some major modifications on the whole structure and I am not sure if this is better than the version in #80.
- Add table of contents.
- Separate section for quick start.
- Divide `Getting start` into three sections: `Env`, `Build`, and `Run`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,230 +7,205 @@
...
-Or build your own CA and TA:
+### Build & Install
 
-```
-$ make -C examples/[YOUR_APPLICATION]
-```
+To build the project, the Rust environment and several related submodules are required.
 
-#### 5. Run Rust applications
+If you are working without QEMUv8, by default, the `OPTEE_DIR` is
```
Comment (DemesneGH) [inline README.md:106]: Suggest removing "If you are working without QEMUv8".

-> By default, xxxxxxx

File: README.md [hunk 2]
```diff
@@ -7,230 +7,205 @@
...
+export OPTEE_DIR=[YOUR_OPTEE_DIR]
 ```
 
-Note: the path `/project/root/dir/` should be replaced as the root directory of
-your local project "Teaclave TrustZone SDK". Copy all the Rust examples or your
-own applications to the shared folder:
+Note: your OPTEE root directory should have `build/`, `optee_os/` and
+`optee_client/` as sub-directory.
+
+Run the script as follows to install the Rust environment and initialize submodules:
```
Comment (DemesneGH) [inline README.md:122]: Add some bullets or numbers to distinguish the different steps, because the notes inside steps may be confusing.

e.g.
1. Set the OP-TEE directory
    ...
2. Install the Rust environment and initialize submodules
    ...
3. Build OP-TEE libraries

Comment (DemesneGH) [inline README.md:122]: Add some bullets or numbers to distinguish the different steps, because the notes inside steps may be confusing.

e.g.
1. Set the OP-TEE directory
    ...
2. Install the Rust environment and initialize submodules
    ...
3. Build OP-TEE libraries

**Review by DemesneGH — CHANGES_REQUESTED** [review]
Thanks!
I prefer the restructured one. Besides the comments, don't forget to wrap all the long lines at 80 characters.

File: README.md [hunk 3]
```diff
@@ -7,230 +7,212 @@
...
-Before building built-in rust examples, you need to build OP-TEE libraries using:
-
-``` sh
-$ make optee
-```
+Alternatively, you can use a docker container built with our
+[Dockerfile](Dockerfile).
 
-Run this command to build all Rust examples:
+2. After installing dependencies or building the Docker image, fetch the source code from the official GitHub repository:
```
Comment (DemesneGH) [inline README.md:95]: rewrap this line

**Comment by DemesneGH** [comment]
LGTM. @mssun Could you help to review this documentation? Thanks!

**Comment by mssun** [comment]
@DemesneGH, I'm okay with the changes. Please squash into one commit and merge. Thanks!

File: optee-teec/Cargo.lock [hunk 4]
```diff
@@ -1,505 +1,505 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
+version = 3
+
```
Comment (DemesneGH) [inline optee-teec/Cargo.lock:4]: @JesseMK Is this file relevant to the current PR? If it was added by mistake, please reset it, thanks!

**Comment by DemesneGH** [comment]
Merged, thanks!
